ocaml/xapi/xapi_vm_lifecycle.ml

```diff
@@ -513,14 +563,7 @@ let update_allowed_operations ~__context ~self =
     | None -> op :: accu
     | _    -> accu
   in
-  let allowed =
-    List.fold_left check []
-      [`snapshot; `copy; `clone; `revert; `checkpoint; `snapshot_with_quiesce;
-       `start; `start_on; `pause; `unpause; `clean_shutdown; `clean_reboot;
-       `hard_shutdown; `hard_reboot; `suspend; `resume; `resume_on; `export; `destroy;
-       `provision; `changing_VCPUs_live; `pool_migrate; `migrate_send; `make_into_template; `changing_static_range;
-       `changing_shadow_memory; `changing_dynamic_range]
```

Comment:
This is deliberately a subset of VM operations.
We do not want to run the checks for all the operations: only the ones for which it is cheap to check whether they are currently allowed.